src/Process/PointCloudProcessing.js

```diff
@@ -0,0 +1,291 @@
+        const { shouldBeLoaded } = shouldDisplayNode(context, layer, elt);
+
+        const boundingBoxEnabled = context.view.camera.camera3D.layers.mask & layer.bboxes.layers.mask;
+        elt.shouldBeLoaded = shouldBeLoaded;
+
+        if (shouldBeLoaded > 0) {
+            elt.notVisibleSince = undefined;
+
+            // only load geometry if this elements has points
+            if (elt.numPoints > 0) {
+                if (elt.obj) {
+                    elt.obj.material.visible = true;
+                    if (__DEBUG__) {
+                        elt.obj.material.uniforms.density.value = elt.density;
+
+                        if (boundingBoxEnabled) {
+                            if (!elt.obj.boxHelper) {
+                                initBoundingBox(elt, layer);
+                            }
+                            elt.obj.boxHelper.visible = true;
+                            elt.obj.boxHelper.material.color.r = 1 - shouldBeLoaded;
+                            elt.obj.boxHelper.material.color.g = shouldBeLoaded;
+                        }
+                    }
+
+                    elt.obj.geometry.setDrawRange(
+                        0, Math.floor(shouldBeLoaded * elt.obj.geometry.attributes.position.count));
+                    layer.counters.pointCount += elt.obj.realPointCount;
+                    layer.counters.displayedCount += Math.floor(shouldBeLoaded * elt.obj.geometry.attributes.position.count);
+                    elt.obj.material.uniforms.size.value = layer.pointSize;
+                } else if (!elt.promise) {
+                    elt.promise = context.scheduler.execute({
+                        layer,
+                        requester: elt,
+                        view: context.view,
+                        priority: 1.0 / elt.name.length, // surfaceOnScreen,
+                        redraw: true,
+                        isLeaf: elt.childrenBitField == 0,
+                    }).then((pts) => {
+                        elt.obj = pts;
+                        // store tightbbox to avoid ping-pong (bbox = larger => visible, tight => invisible)
+                        elt.tightbbox = pts.tightbbox;
+                        pts.geometry.setDrawRange(
+                            0, shouldBeLoaded * pts.geometry.attributes.position.count);
+
+                        // make sure to add it here, otherwise it might never
+                        // be added nor cleaned
+                        layer.group.add(elt.obj);
+
+                        elt.obj.owner = elt;
+                        elt.promise = null;
+                    });
+                }
+            }
+        } else if (shouldBeLoaded < 0) {
+            // not visible
+            if (!elt.notVisibleSince) {
+                elt.notVisibleSince = Date.now();
+            }
+        }
+
+        if (shouldBeLoaded >= 0.9 && elt.children && elt.children.length) {
+            return elt.children;
+        }
+        return undefined;
```

Comment:
Yes I know it's not needed, just found it more consistent and safer. Not very important though.